File: judge-web/src/core/repository/dto/response/SubmissionPrivateResponseDTO.ts
import { SubmissionStatus } from "@/core/domain/enumerate/SubmissionStatus";
import { Language } from "@/core/domain/enumerate/Language";
import { Attachment } from "@/core/domain/model/Attachment";

export type SubmissionPrivateResponseDTO = {
  id: number;
  problem: {
    id: number;
    title: string;
  };
  member: {
    id: number;
    name: string;
  };
  language: Language;
  status: SubmissionStatus;
  code: Attachment;
  createdAt: string;
  updatedAt: string;
};

File: judge-api/src/main/kotlin/io/leonfoliveira/judge/api/controller/dto/response/SubmissionPrivateResponseDTO.kt
package io.leonfoliveira.judge.api.controller.dto.response

import io.leonfoliveira.judge.core.domain.entity.Submission
import io.leonfoliveira.judge.core.domain.enumerate.Language
import java.time.LocalDateTime

data class SubmissionPrivateResponseDTO(
    val id: Int,
    val member: MemberResponseDTO,
    val problem: ProblemPublicResponseDTO,
    val language: Language,
    val status: Submission.Status,
    val code: AttachmentResponseDTO,
    val createdAt: LocalDateTime,
)

fun Submission.toPrivateResponseDTO(): SubmissionPrivateResponseDTO {
    return SubmissionPrivateResponseDTO(
        id = id,
        member = member.toPrivateResponseDTO(),
        problem = problem.toPublicResponseDTO(),
        language = language,
        status = status,
        code = code.toResponseDTO(),
        createdAt = createdAt,
    )
}
